addon/components/utils/social-media-handle.ts

```diff
@@ -0,0 +1,115 @@
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+import { tracked } from '@glimmer/tracking';
+import Component from '@glimmer/component';
+
+interface UtilsSocialMediaHandleArgs {
+  socialNetwork: string;
+  handle: string;
+  selectorOnly?: boolean;
+  onChange(socialNetwork: string, handle: string, formattedUrl: string): void;
+}
+
+type SocialNetworkData = {
+  name: string;
+  formattedUrl: string;
+};
+
+export const SOCIAL_MEDIA_NETWORKS: SocialNetworkData[] = [
```

Comment:
No Pinterest here?

Reply:
No :) basing this list of what is needed in the specs.
If we need anything else in the future, we'll add the ability to pass a dictionnary to the component